File: 260314/K개 중에 1개를 N번 뽑기/n-permutations-of-k-with-repetition.java
import java.util.Scanner;

public class Main {

    static void comb(int idx, int depth){
        if(depth == n){
            for(int i=0; i<arr.length;i++){
                System.out.print(arr[i]+" ");
            }
            System.out.println();
            return;
        }

        for(int i=1; i<=k;i++){
            arr[depth] = i;
            comb(i,depth+1);
        }

    }
    static int k,n;
    static int[] arr;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        k = sc.nextInt();
        n = sc.nextInt();
        // Please write your code here.
        arr = new int[n];
        comb(0,0);


    }
}

File: Codetree/260513/K개 중에 1개를 N번 뽑기/n-permutations-of-k-with-repetition.py
K, N = map(int, input().split())

# Please write your code here.

result = []
def choose(curr_idx):

    if curr_idx == N+1:
        print(*result)
        return

    for i in range(1,K+1):

        result.append(i)
        choose(curr_idx+1)
        result.pop()
        

    return
choose(1)
